### PR TITLE
Don't compile as much stuff

### PR DESCRIPTION
**Pull request by colin-grapl**
### Which issue does this PR correspond to?
None

### What changes does this PR make to Grapl? Why?
1. Statically links to zstd in the `kafka` crate
2. Statically links to openssl in the `kafka` crate
3. Syncs the opentelemetry versions in `uid-allocator` to be the same as elsewhere

### How were these changes tested?
No additional testing

### CODE REVIEW

**Review by jgrillo-grapl — APPROVED** [review]
LGTM

File: src/rust/Dockerfile [hunk 1]
```diff
@@ -22,13 +22,13 @@ RUN --mount=type=cache,target=/var/lib/apt/lists,sharing=locked,id=rust-base-apt
...
         jq=1.6-2.1 \
         unzip=6.0-26+deb11u1 \
-    && apt-get install --yes --no-install-recommends \
         build-essential=12.9 \
         cmake=3.18.4-2+deb11u1 \
         libssl-dev=1.1.1n-0+deb11u3 \
         perl=5.32.1-4+deb11u2 \
         pkg-config=0.29.2-1 \
-        tcl=8.6.11+1
+        tcl=8.6.11+1 \
```
Comment (christophermaier) [inline src/rust/Dockerfile:30]: I'm always surprised by this `tcl` dependency :joy:

**Review by christophermaier — APPROVED** [review]
👍

**Comment by colin-grapl** [comment]
Sadly this didn't work. I assumed the libs were getting statically compiled but services fail at runtime trying to load the .so, indicating that I was incorrect about that.

**Pull request closed without merging**